### PR TITLE
Add elm exercise 1-a

### PR DESCRIPTION
**Pull request by ceddlyburge**
Hi There

This is Elm exercise A, part 1.

It was originally a bit more complex, but has become simpler after we received the feedback which you can see at [Issue 1](https://github.com/ceddlyburge/research_experiment_1/issues/1#issue-537890139).

This should take an experienced Elm programmer around 10 minutes.

I think there is some potential to add some help to the readme to make it easier / quicker, without guiding the user to a solution. I had to look up `String.fromInt` for example, which took time, and is basically incidental to the exercise.

### CODE REVIEW

File: exercises/elm-1-a/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## Instructions
+
+Your task is to take a list of scorers for a football game, and aggregate any duplicates. The aggregated list should represent each scorer once with the number of goals after their name in brackets, if they have scored multiple goals.
+
+The aggregated list should be ordered alphabetically.
+
+For example, if the initial list of scorers is
+
+["Lisa", "Anna", "Lisa"]
```
Comment (ErikSchierboom) [inline exercises/elm-1-a/README.md]: Maybe add a code block block around it?

File: exercises/elm-1-a/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Then the aggregated list should look like this. 
+
+[ "Anna", "Lisa (2)" ]
```
Comment (ErikSchierboom) [inline exercises/elm-1-a/README.md]: Maybe add a code block block around it?

File: exercises/elm-1-a/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Introduction
+
+This is Part 1 of our research into how people write Elm differently. This exercise should take 10-15 minutes. There is no right or wrong way to approach it - just do what feels the most natural to you. Feel free to use books, Google or Stack Overflow, just like you would if you were programming normally, but please don't actively look for a solution to the problem as this will negatively affect the research.
+
+## Instructions
```
Comment (ErikSchierboom) [inline exercises/elm-1-a/README.md]: Maybe be explicit about which method the student is expected to define? See e.g. the [C# instructions](https://github.com/exercism/research_experiment_1/blob/master/exercises/csharp-1-a/README.md#instructions).

File: exercises/elm-1-a/README.md (hunk 1 above)
Comment (ErikSchierboom) [inline exercises/elm-1-a/README.md]: A small suggestion could be to show an actual function call here:

```elm
aggregateScorers ["Lisa", "Anna", "Lisa"]
```

You could also show the expected result:

```elm
aggregateScorers ["Lisa", "Anna", "Lisa"]
-- Returns: [ "Anna", "Lisa (2)" ] 
```

File: exercises/elm-1-a/tests/Tests.elm [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    describe "AggregateScorers"
+        [ test "doesn't change a single scorer" <|
+            \() ->
+                aggregateScorers [ "Anna" ]
+                |> Expect.equal [ "Anna" ]
+        , test "doesn't change multiple unique scorers" <|
+            \() ->
+                aggregateScorers [ "Anna", "Lisa" ]
+                |> Expect.equal [ "Anna", "Lisa" ]  
+        , test "converts duplicates in to goals" <|
```
Comment (ErikSchierboom) [inline exercises/elm-1-a/tests/Tests.elm]: Maybe add a test case for the example mentioned before, where there are multiple scorers and one of them has scored more than once?

File: exercises/elm-1-a/.meta/config.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      "name": "doesn't change a single scorer",
+      "cmd": "aggregateScorers [ \"Anna\" ]"
+    },
+    {
+      "name": "doesn't change multiple unique scorers",
+      "cmd": "aggregateScorers [ \"Anna\", \"Lisa\" ]"
+    },
+    {
+      "name": "converts duplicates in to goals",
```
Comment (ErikSchierboom) [inline exercises/elm-1-a/.meta/config.json]: I'm not entirely sure, but I think this should be:

```suggestion
      "name": "converts duplicates into goals",
```

File: exercises/elm-1-a/tests/Tests.elm (hunk 4 above)
Comment (ErikSchierboom) [inline exercises/elm-1-a/tests/Tests.elm]: I'm not entirely sure, but I think this should be:

```suggestion
        , test "converts duplicates into goals" <|
```

**Comment by ceddlyburge** [comment]
Hi Erik, thanks for the comments! The plan for part 2 is to include the time at which the players scored the goals, and include this in the aggregate list. 

For example:

```elm
aggregateScorers ["Lisa 21", "Anna 37", "Lisa 76"]
-- Returns: [ "Lisa (21)", "Anna (37, 76)" ]
```
Elm-1-b is basically ready as well, but Matthieu is away doing christmas. I expect he will create a Pull Request fairly soon.

Cheers, Cedd

**Pull request merged**